.github/workflows/test.yml

```diff
@@ -36,6 +36,7 @@ jobs:
           - macos-12
           - windows-2019
         cmake-version:
+          - 3.31.0
           - 3.20.0
           - 3.10.2
```

Comment:
We should just set this to 3.16.3, which is the version used in ubuntu 20.04, then we could just update the minimum version to 3.16 in the test files.